app/routes.py

```diff
@@ -57,10 +57,10 @@ def login():
     if form_log_in.validate_on_submit():
         flash('Login requested for user {}, remember_me={}'.format(
             form_log_in.username.data, form_log_in.remember_me.data))
-        user = models.User.query.filter_by(username=form_log_in.username.data).first()
+        user = utils.find_user_by_username(form_log_in.username.data)
         if user is None or not user.check_password(form_log_in.password.data):
             flash('invalid username or password')
-            return redirect('/login/')
+            return render_template('login.html', title='Log In', form=form_log_in, invalid=True)
```

Comment:
you should display error messages to the user like this via flash(). Here's a good example: https://stackoverflow.com/questions/40949746/how-to-display-flashing-message-without-reloading-the-page-in-flask